enhancements/network/ovn-multi-external-gateway.md

```diff
@@ -0,0 +1,178 @@
+* `k8s.ovn.org/routing-external-gws`
+* `k8s.ovn.org/external-gw-pod-ips`
+
+Pod annotations:
+* `k8s.ovn.org/routing-namespaces`
+* `k8s.ovn.org/routing-network`
+* `k8s.ovn.org/bfd-enabled`
+
+The implementation will also include a new controller that will watch cluster wide for instances of this CRD and apply the changes to the OVN network configuration as it is currently done by the annotation based logic.
+
+The controller will use the OpenShift release version as well as the existence of the annotations in the namespace and pods to apply the changes to the network configuration:
+
+| **OpenShift version** | **Annotations exist** | **CR exists** | **Action**                                        |
+|-----------------------|-----------------------|---------------|---------------------------------------------------|
+| 4.13                  | No                    | Yes           | Use CR information to configure external gateways |
+| 4.13                  | Yes                   | Yes           | Use annotations to configure external gateways    |
+| 4.13                  | Yes                   | No            | Use annotations to configure external gateway     |
+| 4.13                  | No                    | No            | Nothing to do                                     |
+| 4.14                  | No                    | Yes           | Use CR information to configure external gateways |
+| 4.14                  | Yes                   | Yes           | Use CR information to configure external gateways |
+| 4.14                  | Yes                   | No            | Use annotations to configure external gateways    |
+| 4.14                  | No                    | No            | Nothing to do                                     |
+
+The current annotation based logic will be enhanced as part of this change to support the logic described in the previous table.
+
+### Workflow Description
+The workflow follows the behavior of a namespaced CR.
+
+
+### API Extensions
+
+A new namespaced-scoped CRD is created to represent the multiple external gateways configuration. The `spec` structure exposes the following fields:
+* `staticGateways` contains a slice of IPv4 or IPv6 addresses that are configured as static external gateways. It maps to the `k8s.ovn.org/routing-external-gws` annotation.
+* `routingNamespaces` contains a slice of the namespaces linked to this gateway configuration. It maps to the `k8s.ovn.org/routing-namespaces` annotation.
+* `podGateways` allows defining the selector and configuration criteria for the gateway pods. This configuration contains the following fields:
+  * `network` references the multus network configuration that matches the same name in the pod that will be used for external traffic. If the field is empty or undefined, the pod's IP will be used instead. This matches the current behavior used with annotations. This field maps to the annotation `k8s.ovn.org/routing-network`.
```

Comment:
Agreed, I need to think about it. In essence we're looking at two possibilities: 
* retrieve the IPs from a network reference from the multus config annotation.
* retrieve the IPs from the node's network interface, assuming that the pod has `hostNetwork` is set to true.

Reply:
can you give an example of what you are thinking?

Reply:
It was more about providing two fields:
* One that determines if the source of the IPs was primary (host network) or secondary (multus managed).
* For the multus case, it provided the network name. In case of the primary network, this field would not be used.